command/flags/connect.go

```diff
@@ -0,0 +1,79 @@
+package flags
+
+import (
+	"flag"
+	"os"
+
+	"github.com/hashicorp/consul/api"
+)
+
+type ConnectFlags struct {
+	proxyID    StringValue
+	proxyToken StringValue
+	sidecarFor StringValue
+}
+
+func (f *ConnectFlags) Init() *flag.FlagSet {
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
+
+	fs.Var(&f.proxyID, "proxy-id",
+		"The proxy's ID on the local agent. This can also be specified via the "+
+			"CONNECT_PROXY_ID environment variable.")
+
+	fs.Var(&f.proxyToken, "proxy-token",
+		"ACL token to use in the request. This can also be specified via the "+
+			"CONNECT_PROXY_TOKEN environment variable. If unspecified, the query will "+
+			"default to the token of the Consul agent at the HTTP address.")
+
+	fs.Var(&f.sidecarFor, "sidecar-for",
+		"The ID of a service instance on the local agent that this proxy should "+
+			"become a sidecar for. It requires that the proxy service is registered "+
+			"with the agent as a connect-proxy with Proxy.DestinationServiceID set "+
+			"to this value. If more than one such proxy is registered it will fail. "+
+			"This can also be specified via the CONNECT_SIDECAR_FOR environment variable.")
+
+	return fs
+}
+
+func (f *ConnectFlags) ProxyID() string {
+	return f.proxyID.String()
+}
+
+func (f *ConnectFlags) SetProxyID(v string) error {
+	return f.proxyID.Set(v)
+}
+
+func (f *ConnectFlags) SidecarFor() string {
+	return f.sidecarFor.String()
+}
+
+func (f *ConnectFlags) APIClient() (*api.Client, error) {
```

Comment:
@mkeeler Currently I don't believe this is actually called anywhere, so the call to `LoadFromEnv` in here never happens and the env vars would never be loaded - where/how should this actually happen? For reference, the usage of `HTTPFlags` looks like https://github.com/hashicorp/consul/blob/b53b98fa26c2d80609ea1ac4f4fd83ee6af53f6a/command/connect/proxy/proxy.go#L157-L162

Reply:
Wasn't quite sure what I was doing here so ended up reverting this chunk of code back to something less DRY.